Dockerfile

```
@@ -1,19 +1,36 @@
-# --- builder ---
-FROM golang:1.20.6-alpine3.17 as builder
-LABEL stage=builder
-RUN apk add git
-WORKDIR /build
+# syntax=docker/dockerfile:1
 
-COPY go.* ./
-RUN go mod download
+ARG app_dir="/home/go/app"
 
-COPY . ./
-ARG BUILD_STRING=pretendo.friends.docker
-RUN go build -ldflags "-X 'main.serverBuildString=${BUILD_STRING}'" -v -o server
 
-# --- runner ---
-FROM alpine:3.17 as runner
-WORKDIR /build
+# * Building the application
+FROM golang:1.22-alpine3.20 AS build
```

The build is currently failing due to an outdated version of Docker:

`go: go.mod requires go >= 1.23.0 (running go 1.22.12; GOTOOLCHAIN=local)`

Please update this Dockerfile to use 1.23

```suggestion
FROM golang:1.23.6-alpine AS build
```